modules/javafx.graphics/src/main/native-glass/gtk/PlatformSupport.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include "PlatformSupport.h"
+#include "glass_general.h"
+#include <gtk/gtk.h>
+
+namespace
+{
+    void putColor(JNIEnv* env, jobject prefs, GtkStyle* style, const char* lookupColorName, const char* prefColorName) {
+        GdkColor color;
+        if (gtk_style_lookup_color(style, lookupColorName, &color)) {
+            env->CallObjectMethod(prefs, jMapPut,
+                env->NewStringUTF(prefColorName),
+                env->CallStaticObjectMethod(
+                    jColorCls, jColorRgb,
+                    (int)(CLAMP((double)color.red / 65535.0, 0.0, 1.0) * 255.0),
+                    (int)(CLAMP((double)color.green / 65535.0, 0.0, 1.0) * 255.0),
+                    (int)(CLAMP((double)color.blue / 65535.0, 0.0, 1.0) * 255.0),
+                    1.0));
+        }
+
+        CHECK_JNI_EXCEPTION(env);
+    }
+
+    void putString(JNIEnv* env, jobject preferences, const char* name, const char* value) {
+        env->CallObjectMethod(preferences, jMapPut,
+            env->NewStringUTF(name),
+            env->NewStringUTF(value));
+    }
+}
+
+PlatformSupport::PlatformSupport(JNIEnv* env, jobject application)
+    : env(env), application(env->NewGlobalRef(application)), preferences(NULL) {}
+
+PlatformSupport::~PlatformSupport() {
+    env->DeleteGlobalRef(application);
+
+    if (preferences) {
+        env->DeleteGlobalRef(preferences);
+    }
+}
+
+jobject PlatformSupport::collectPreferences() const {
+    jobject prefs = env->NewObject(jHashMapCls, jHashMapInit);
+    if (EXCEPTION_OCCURED(env)) return NULL;
+
+    GtkStyle* style = gtk_style_new();
```

Comment:
It's not specified to do so. Internally, it will call `g_object_new`, which is also not specified to return null. I've found some evidence on some gnome.org blog that it will not ever return null, but I haven't looked further than that.

Reply:
I suspect that any object allocation method, such as `g_object_new` could return null if memory is exhausted, so a NULL check seems like a good idea.